File: JavaComparisonModule/NetworkEmulationJava/src/main/java/com/github/insanusmokrassar/KotlinVSJava/JavaComparisonModule/NetworkEmulationJava/NetworkImpl.java
package com.github.insanusmokrassar.KotlinVSJava.JavaComparisonModule.NetworkEmulationJava;

public class NetworkImpl implements Network {
    private final long defaultDelay;

    public NetworkImpl(long defaultDelay) {
        this.defaultDelay = defaultDelay;
    }

    public NetworkImpl() {
        this(1000L);
    }

    @Override
    public int request(int data) {
        try {
            Thread.sleep(defaultDelay);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        return data;
    }
}

File: KotlinComparisonModule/NetworkEmulationKotlin/src/main/kotlin/com/github/insanusmokrassar/KotlinVSJava/KotlinComparisonModule/NetworkEmulationKotlin/NetworkImpl.kt
package com.github.insanusmokrassar.KotlinVSJava.KotlinComparisonModule.NetworkEmulationKotlin

import kotlinx.coroutines.delay

class NetworkImpl(
    private val defaultDelay: Long = 1000L
) : Network {
    override suspend fun makeRequest(data: Int): Int {
        delay(defaultDelay)
        return data
    }
}